TESTS/mbed_platform/minimal-printf/compliance/main.cpp

```diff
@@ -51,6 +51,8 @@ static control_t test_printf_d(const size_t call_count)
 
     /*************************************************************************/
     /*************************************************************************/
+#if !defined(__NEWLIB_NANO)
+    // The format specifier %hhd is not supported by Newlib-Nano
```

Comment:
That doesn't make sense to me. These tests are for minimal-printf not for standard printf.  I'm assuming they are being run in all cases even if the minimal-printf is not present and that's the problem. The thing is this change will possibly hide issues as we won't execute tests in usecase with minimal-printf and minimal clibs. I think the correct solution is to exclude all minimal-printf tests if minimal-printf is not used, but run them with newlib-nano + minimal-printf as the newlib-nano limitation won't be hit.

Reply:
The user has the choice to swap the standard printf by minimal-printf ; `mbed_printf` is a test API that is always present (it calls `mbed_minimal_formatted_string` which is the minimal printf implementation). 
But you are right that the limitation is a newlib-nano limitation and the test could check against a predefined value to avoid calling printf.

Reply:
It's also a waste of time in the CI (even if not huge one) to test the stdlib. Sure it possible to find bugs there but it's rather rare.